src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -0,0 +1,43 @@
+// <copyright file="InMemoryExporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry;
+
+namespace OpenTelemetry.Exporter
+{
+    public class InMemoryExporter<T> : BaseExporter<T>
+        where T : class
+    {
+        private readonly ICollection<object> trait;
+
+        public InMemoryExporter(InMemoryExporterOptions options = null)
```

Comment:
You could do InMemoryExpoterOptions<T> here so the list/collection could be strongly typed (ICollection<T> Collection { get; set; }).

Reply:
I did this intentional - trying to get feedback: in this way we can have a single collection used by different types of exporters, I've put a demo in the README.md. Might be over-enginering?

Reply:
FYI, this is what I was trying:


```csharp
var list = new List<object>();
var activityExporter = new InMemoryExporter<Activity>(new InMemoryExporterOptions { Collection = list });
var logExporter = new InMemoryExporter<LogRecord>(new InMemoryExporterOptions { Collection = list });
// do something about the synthesized data.
```

Reply:
Ah, cool. I think a typed version would be more useful, more often, but given this is for unit testing primarily (?), a cast is probably fine.

Reply:
I've added contravariance, see if you like it @CodeBlanch.